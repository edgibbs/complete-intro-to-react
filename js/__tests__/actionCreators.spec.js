// @flow

import {setSearchTerm, addAPIData} from '../actionCreators';

describe('#setSearchTerm', () => {
  it('returns an action object', () => {
    expect(setSearchTerm('New York')).toMatchSnapshot();
    expect(setSearchTerm('New York')).toEqual({
      type: 'SET_SEARCH_TERM',
      payload: 'New York'
    });
  });
});

describe('#addAPIData', () => {
  it('returns an action object', () => {
    expect(
      addAPIData({
        title: 'Stranger Things',
        imdbID: '12',
        rating: '8.6'
      })
    ).toEqual({
      type: 'ADD_API_DATA',
      payload: {title: 'Stranger Things', imdbID: '12', rating: '8.6'}
    });
  });
});
