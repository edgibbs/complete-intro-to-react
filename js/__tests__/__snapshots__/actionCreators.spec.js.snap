// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#setSearchTerm returns an action object 1`] = `
Object {
  "payload": "New York",
  "type": "SET_SEARCH_TERM",
}
`;
